db/view/view.cc

```diff
@@ -332,14 +378,33 @@ static query::partition_slice make_partition_slice(const schema& s) {
             std::move(opts));
 }
 
+view_builder::build_step& view_builder::get_or_create_build_step(utils::UUID base_id) {
+    auto it = _base_to_build_step.find(base_id);
+    if (it == _base_to_build_step.end()) {
+        auto base = _db.find_column_family(base_id).shared_from_this();
+        auto p = _base_to_build_step.emplace(base_id, build_step{base, make_partition_slice(*base->schema())});
+        // Iterators could have been invalidated if there was rehashing, so just reset the cursor.
+        _current_step = p.first;
+        it = p.first;
+    }
+    return it->second;
+}
+
 bool matches_view_filter(const schema& base, const view_info& view, const partition_key& key, const clustering_row& update, gc_clock::time_point now) {
+    std::vector<bytes> exploded_pk = key.explode();
+    std::vector<bytes> exploded_ck = update.key().explode();
+    auto slice = make_partition_slice(base);
+
+    data_query_result_builder builder(base, slice, query::result_options::only_result(), query::result_memory_accounter{10*1024*1024});
+    builder.consume_new_partition(dht::decorate_key(base, key));
+    builder.consume(clustering_row(base, update), row_tombstone{}, update.is_live(base, tombstone{}, now));
+    builder.consume_end_of_partition();
+    auto result = builder.consume_end_of_stream();
+    view_filter_checking_visitor visitor(base, view,exploded_pk, exploded_ck);
```

Comment:
In theory you could have folded these two visitors into one, but I totally understand why not.